packages/lodestar/src/chain/bls/multithread/index.ts

```diff
@@ -136,6 +143,22 @@ export class BlsMultiThreadWorkerPool implements IBlsVerifier {
   }
 
   async verifySignatureSets(sets: ISignatureSet[], opts: VerifySignatureOpts = {}): Promise<boolean> {
+    if (opts.useMainThread && !this.blsVerifyAllMultiThread) {
+      const startNs = process.hrtime.bigint();
+      const isValid = verifySignatureSetsMaybeBatch(
+        sets.map((set) => ({
+          publicKey: getAggregatedPubkey(set),
+          message: set.signingRoot.valueOf() as Uint8Array,
+          signature: set.signature,
+        }))
+      );
+
+      const endNs = process.hrtime.bigint();
+      this.metrics?.blsTime.mainThreadDurationInThreadPool.observe(Number(endNs - startNs) / 1e9);
```

Comment:
You can use the timer function which does calls the process time for you. Also wrap in try {} finally {} in case verifySignatureSetsMaybeBatch rejects

```ts
const timer = metrics.startTimer()
try {
  return fn()
} finally {
  timer()
}
```